addons/mail/models/mail_thread.py

```diff
@@ -3401,3 +3403,14 @@ def _get_mail_thread_data(self, request_list):
         if 'suggestedRecipients' in request_list:
             res['suggestedRecipients'] = self._message_get_suggested_recipients()[self.id]
         return res
+
+    def _related_res_partner_default_create_values(self, normalized_email):
```

Comment:
It is a bit strange to have this "normalized" email parameter... as the method simply returns values for a new partner, it is unclear what is the purpose of this "email". As anyway it is used to check if we have to return values or not (aka, is the input matching the record customer email), shouldn't we remove it and let the caller decide if he has to call the method or not ?

Alternatively, if we want to enable a more batch- or broad approach, we could name it "filter_email", and let the implementation deal with it as you do currently. Note that in that case we could think of an override in sms with filter_mobile or filter_phone, to check partners based on mobile or phone numbers. But maybe not now, as it is not part of the spec... just thinking.

Reply:
Yes, the batch approach is a good idea, thanks. The method _get_customer_information returns now a dictionary normalized mail → customer values (for initializing related res_partner).